from_mwt_ds/DataScience/estimate.py

```diff
@@ -0,0 +1,45 @@
+import pandas as pd
+from estimators.bandits import ips, snips, mle, cressieread
+
+def estimate_bucket(bucket: pd.DataFrame, config: dict) -> dict:
+    if len(bucket) == 0 or config == {}:
+        return {}
+    else:
+        for policy_name, estimator_list in config["policies"].items():
+            estimator_obj = None
+            for estimator in estimator_list:
+                if estimator == 'ips':
+                    estimator_obj = ips.Estimator()
+                elif estimator == 'snips':
+                    estimator_obj = snips.Estimator()
+                elif estimator == 'mle':
+                    estimator_obj = mle.Estimator()
+                elif estimator == 'cressieread':
+                    estimator_obj = cressieread.Estimator()
+                else:
+                    raise("Estimator not found.")
+
+                for index, row in bucket.iterrows():
```

Comment:
remove the `index` here, we don't need it

Reply:
I was poking around and it looks like if I use iterrows() I have to unpack index somehow from the examples I saw on SO. Still poking around for how I could just do one; the code seems to fail otherwise.

Reply:
I was reading the second answer from this thread: https://stackoverflow.com/questions/16476924/how-to-iterate-over-rows-in-a-dataframe-in-pandas seems iterrow() is not an efficient way to loop through the dataframe.

Reply:
yeah, I was reading that as well.